docs/ctapipe_api/image/extractor.rst

```diff
@@ -1,10 +1,75 @@
 .. _image_charge_extractors:
 
-==================================
-Charge Extraction (cube to image) 
-==================================
+#########################################
+Image Extraction (waveform cube to image)
+#########################################
 
-Extracts charge and time per pixel from the waveform.
+The main operation in the event processing from DL0 to Dl1 is the reducing of the
+waveforms into per-pixel signal-summarising parameters, from which image analysis
+can be performed to reconstruct the Cherenkov shower. This summary is
+performed for three reasons:
+
+* The amount of useful information is arguably a small fraction of
+  the total waveform volume (across all pixels in the Cherenkov camera).
+
+* There is typically at most one single Cherenkov signal pulse per waveform.
+
+* It is easier to perform further image analysis techniques on 2-dimensional
+  images. This is a much more common image processing situation, for which many
+  standard techniques exist across data science.
+
+This operation is performed by an `ImageExtractor`. This class summarises the
+signal contained in the waveform with two parameters:
+
+* The signal's charge
+
+* The position (time) of the signal with the waveform.
+
+
+************************
+Image Extraction Methods
+************************
+
+The field of signal processing is very widespread, and many common
+techniques are applicable to the waveforms from Cherenkov cameras. The large
+range of options is therefore reflected in the design of the `ImageExtractor`
+in *ctapipe*. `ImageExtractor` itself is an abstract class. Subclasses of
+`ImageExtractor` define the approach for extracting the charge and the time.
+Typical extraction techniques involve two stages:
+
+* Peak finding, to identify the most probable location for the signal
+
+* Integration, to extract the charge with a window that is large enough to
+  capture the signal, but small enough that it minimises the amount of noise
+  (e.g. Night Sky Background, electronic...) included.
+
+However, an `ImageExtractor` is not forced to conform to these stages, and is
+free to utilise any information from the camera description (such as reference
+pulse shape or neighbouring pixels).
+
+The most important criteria in selecting and configuring an `ImageExtractor` is
+the Charge/Intensity Resolution - how well the extracted charge matches the true
+charge. The best `ImageExtractor` for one camera type may not be the same for
+another.
+
+See below for the currently existing `ImageExtractor` classes.
+
+
+***********************
+Units and Normalisation
+***********************
+
+An `ImageExtractor` should preserve the units of the waveform samples when
+extracting the charge. I.e. if the samples are already calibrated into
+photoelectrons, then the charge reported by the `ImageExtractor` should be in
+photoelectrons. The simple `ImageExtractors` (e.g. `LocalPeakWindowSum`)
```

Comment:
This is a very general comment but it seems to apply here.

How can samples be photo electrons? The conversion from the sum or integral of the waveforms is always dependent on the method used and its settings.

There needs to be a conversion from waveform integral / sum or whatever the extractor calculates (could also just use the maxium or width or whatever) to  photo electrons.

Shouldn't we just stop stalking about a waveform being in units of photo electrons?
Or at least have a well defined definition of what that means?

Reply:
In the SimtelEventSource, the dc_to_pe factor is applied to the samples.

If the waveform from simtelarray only contained an X p.e. signal, no noise or NSB, then the sum of all the samples would equal X. Therefore technically the waveform samples are in units of photoelectrons.

In the case of a window summing a subset of the signal pulse, the integration_correction will scale the summed samples by the amount of signal not included in the sum, to ensure the same answer is still returned. Therefore ensuring the units of the charge are still in p.e.

I accept that the same may not be true from a different EventSource - the units may not be in units of photoelectrons. In this case, whatever units the waveform samples are provided in, then the resulting extracted charge will have the same units. A further calibration may be needed to convert the charge to photoelectrons (which is possible in the CameraCalibrator). 

I have tried to convey this in the documentation, giving the simtelarray -> photoelectrons case as an example.

Reply:
There is no integral over time performed in the charge extraction. The "sampling" of the waveform has already performed this integral of the continuous readout into the waveform samples.

Reply:
I suppose it's not really true for FlashADCs (they do not integrate, but really sample, and FlashCam has a non-linear amplifier, so it's not PE).